spec/outputs/newrelic_spec.rb

```diff
@@ -48,6 +49,48 @@
               })).to have_been_made
     end
   end
+
+  context "check https connection scheme" do
+    it "uses https by default" do
+      stub_request(:any, base_uri).to_return(status: 200)
+
+      event = LogStash::Event.new({:message => "Test message" })
+      @newrelic_output.multi_receive([event])
+
+      wait_for(a_request(:post, base_uri)
+        .with(headers: {
+                "X-License-Key" => license_key,
+                "X-Event-Source" => "logs",
+                "Content-Encoding" => "gzip",
+              })).to have_been_made
+
+      # Check if the requests were made using HTTPS
+      expect(WebMock).to have_requested(:post, base_uri).with { |req| req.uri.scheme == 'https' }
```

Comment:
Can you please also check that the requested port is the default 443, in this case (as `base_uri` doesn't explicitly define the port)?

Reply:
Done